sdk/core/Azure.Core/api/Azure.Core.net461.cs

```diff
@@ -569,3 +570,29 @@ public abstract partial class ObjectSerializer
         public virtual System.Threading.Tasks.ValueTask<System.BinaryData> SerializeAsync(object? value, System.Type? inputType = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
     }
 }
+namespace Azure.Messaging
+{
+    [System.Text.Json.Serialization.JsonConverterAttribute(typeof(Azure.Messaging.CloudEventConverter))]
+    public partial class CloudEvent
+    {
+        public CloudEvent(string source, string type, object jsonSerializable, System.Type dataSerializationType = null) { }
+        public CloudEvent(string source, string type, System.ReadOnlyMemory<byte> data, string dataContentType) { }
+        public string DataContentType { get { throw null; } set { } }
+        public string DataSchema { get { throw null; } set { } }
+        public System.BinaryData EventData { get { throw null; } }
+        public System.Collections.Generic.IDictionary<string, object> ExtensionAttributes { get { throw null; } }
+        public string Id { get { throw null; } set { } }
+        public string Source { get { throw null; } set { } }
+        public string Subject { get { throw null; } set { } }
+        public System.DateTimeOffset? Time { get { throw null; } set { } }
+        public string Type { get { throw null; } set { } }
+        public static Azure.Messaging.CloudEvent[] Parse(string requestContent) { throw null; }
+    }
+    public partial class CloudEventConverter : System.Text.Json.Serialization.JsonConverter<Azure.Messaging.CloudEvent>
+    {
+        public CloudEventConverter() { }
+        public Azure.Core.Serialization.ObjectSerializer DataSerializer { get { throw null; } set { } }
```

Comment:
Why isn't this a ctor parameter?

Reply:
I initially made it an optional ctor parameter, but that broke the default serialization because there was no longer a parameterless ctor. I can add two separate ctors though.

Reply:
I think we should do 2 constructors.

Reply:
There is a discussion as to whether it would be better to leave this internal and have a `GetJsonConverter` method on `CloudEvent` that takes an `ObjectSerializer`. Based on the outcome of that discussion this may become moot.